clang/test/CodeGenSYCL/function-attrs.cpp

```diff
@@ -29,21 +29,10 @@ int foo() {
 }
 
 template <typename Name, typename Func>
-__attribute__((sycl_kernel)) void kernel_single_task(const Func &kernelFunc) {
+__attribute__((sycl_kernel_entry_point(Name))) void kernel_single_task(const Func &kernelFunc) {
   kernelFunc();
 }
 
-// CHECK-LABEL: define dso_local noundef i32 @main(
-// CHECK-SAME: ) #[[ATTR0]] {
-// CHECK-NEXT:  entry:
-// CHECK-NEXT:    [[RETVAL:%.*]] = alloca i32, align 4
-// CHECK-NEXT:    [[REF_TMP:%.*]] = alloca [[CLASS_ANON:%.*]], align 1
-// CHECK-NEXT:    [[RETVAL_ASCAST:%.*]] = addrspacecast ptr [[RETVAL]] to ptr addrspace(4)
-// CHECK-NEXT:    [[REF_TMP_ASCAST:%.*]] = addrspacecast ptr [[REF_TMP]] to ptr addrspace(4)
-// CHECK-NEXT:    store i32 0, ptr addrspace(4) [[RETVAL_ASCAST]], align 4
-// CHECK-NEXT:    call spir_func void @_Z18kernel_single_taskIZ4mainE11fake_kernelZ4mainEUlvE_EvRKT0_(ptr addrspace(4) noundef align 1 dereferenceable(1) [[REF_TMP_ASCAST]]) #[[ATTR1]]
-// CHECK-NEXT:    ret i32 0
-//
```

Comment:
It isn't clear to me what this test is intended to validate. Given the name, I'm guessing the important checks are related to the `// CHECK: attributes` lines below.

Hmm, I did a little spelunking and now I think the intent of the test was broken by commit 2dc6579f6fb12470559a68886c2a4aecaa8495dd; the test no longer validates that `bar()` is emitted with the `convergent` and `nounwind` attributes. The test was introduced by commit 6b868139458d258c1ed4c0279e8f4374556c1c1e and the commit message specifically indicates the intent was to check that the `nounwind` attribute is present. We should fix this, but perhaps as part of a separate PR.

Reply:
I fixed this. I just changed to check for ATTR0 which now checks that the `nounwind` attribute is generated